Source/p_inter.c

```diff
@@ -695,26 +695,37 @@ static void P_KillMobj(mobj_t *source, mobj_t *target)
 #ifndef MBF_STRICT
       // For compatibility with PrBoom+ complevel 11 netgame
       else
-        if (
-              (target->flags & MF_COUNTKILL) &&
-              (demo_version >= 203) && netgame && !deathmatch &&
-              !(
-                 target->lastenemy &&
-                 target->lastenemy->health > 0 &&
-                 target->lastenemy->player
-               )
-           )
+        if (netgame && !deathmatch && (target->flags & MF_COUNTKILL))
+        {
+          if (target->lastenemy && target->lastenemy->health > 0 &&
+              target->lastenemy->player)
+          {
+              target->lastenemy->player->killcount++;
+          }
+          else
           {
-            int i;
-            for (i = 0; i < MAXPLAYERS; ++i)
+            int i, playerscount;
+
+            for (i = 0, playerscount = 0; i < MAXPLAYERS; ++i)
             {
               if (playeringame[i])
               {
-                P_Random(pr_friends);
-                break;
+                if (demo_version < 203)
+                {
+                  players[i].killcount++;
+                  break;
```

Comment:
PrBoom+ gives all kills to the first player in vanilla demos. Maybe we can randomize it here too, but with `Woof_Random` function?

Reply:
Yes, okay.